{
  "document_title": "CVE-2019-19274",
  "document_type": "SUSE CVE",
  "document_publisher": {
    "type": "Vendor",
    "contact_details": "security@suse.de",
    "issuing_authority": "SUSE Security Team"
  },
  "document_tracking": {
    "identification": {
      "id": "SUSE CVE-2019-19274"
    },
    "status": "Interim",
    "version": "1",
    "revision_history": {
      "revision": {
        "number": "12",
        "date": "2023-09-07T00:10:21Z",
        "description": "current"
      }
    },
    "initial_release_date": "2021-05-30T14:33:28Z",
    "current_release_date": "2023-09-07T00:10:21Z",
    "generator": {
      "engine": "cve-database/bin/generate-cvrf-cve.pl",
      "date": "2020-12-27T01:00:00Z"
    }
  },
  "document_notes": [
    {
      "text": "CVE-2019-19274",
      "title": "CVE",
      "type": "Summary"
    },
    {
      "text": "typed_ast 1.3.0 and 1.3.1 has a handle_keywordonly_args out-of-bounds read. An attacker with the ability to cause a Python interpreter to parse Python source (but not necessarily execute it) may be able to crash the interpreter process. This could be a concern, for example, in a web-based service that parses (but does not execute) Python code. (This issue also affected certain Python 3.8.0-alpha prereleases.)",
      "title": "Mitre CVE Description",
      "type": "Description"
    },
    {
      "text": "The CVRF data is provided by SUSE under the Creative Commons License 4.0 with Attribution (CC-BY-4.0).",
      "title": "Terms of Use",
      "type": "Legal Disclaimer"
    }
  ],
  "document_references": [
    {
      "type": "Self",
      "url": "https://lists.opensuse.org/opensuse-security-announce/2020-05/msg00001.html",
      "description": "E-Mail link for openSUSE-SU-2020:0567-1"
    },
    {
      "type": "Self",
      "url": "https://lists.opensuse.org/opensuse-security-announce/2020-05/msg00012.html",
      "description": "E-Mail link for openSUSE-SU-2020:0609-1"
    },
    {
      "type": "Self",
      "url": "https://www.suse.com/support/security/rating/",
      "description": "SUSE Security Ratings"
    }
  ],
  "product_tree": {
    "branch": [
      {
        "type": "Product Family",
        "name": "SUSE Package Hub 15 SP1",
        "branch": [
          {
            "type": "Product Name",
            "name": "SUSE Package Hub 15 SP1",
            "full_product_name": {
              "text": "SUSE Package Hub 15 SP1",
              "product_id": "SUSE Package Hub 15 SP1"
            }
          }
        ],
        "full_product_name": {}
      },
      {
        "type": "Product Family",
        "name": "openSUSE Leap 15.1",
        "branch": [
          {
            "type": "Product Name",
            "name": "openSUSE Leap 15.1",
            "full_product_name": {
              "text": "openSUSE Leap 15.1",
              "product_id": "openSUSE Leap 15.1",
              "cpe": "cpe:/o:opensuse:leap:15.1"
            }
          }
        ],
        "full_product_name": {}
      },
      {
        "type": "Product Version",
        "name": "python3-typed-ast-1.3.1-bp151.2.6.1",
        "full_product_name": {
          "text": "python3-typed-ast-1.3.1-bp151.2.6.1",
          "product_id": "python3-typed-ast-1.3.1-bp151.2.6.1"
        }
      },
      {
        "type": "Product Version",
        "name": "python3-typed-ast-1.3.1-lp151.2.6.1",
        "full_product_name": {
          "text": "python3-typed-ast-1.3.1-lp151.2.6.1",
          "product_id": "python3-typed-ast-1.3.1-lp151.2.6.1"
        }
      }
    ],
    "relationship": [
      {
        "product_reference": "python3-typed-ast-1.3.1-bp151.2.6.1",
        "relation_type": "Default Component Of",
        "relates_to_product_reference": "SUSE Package Hub 15 SP1"
      },
      {
        "product_reference": "python3-typed-ast-1.3.1-lp151.2.6.1",
        "relation_type": "Default Component Of",
        "relates_to_product_reference": "openSUSE Leap 15.1"
      }
    ]
  },
  "vulnerability": {
    "notes": [
      {
        "text": "typed_ast 1.3.0 and 1.3.1 has a handle_keywordonly_args out-of-bounds read. An attacker with the ability to cause a Python interpreter to parse Python source (but not necessarily execute it) may be able to crash the interpreter process. This could be a concern, for example, in a web-based service that parses (but does not execute) Python code. (This issue also affected certain Python 3.8.0-alpha prereleases.)",
        "title": "Vulnerability Description",
        "type": "General"
      }
    ],
    "cve": "CVE-2019-19274",
    "product_statuses": [
      {
        "type": "Fixed",
        "product_id": [
          "SUSE Package Hub 15 SP1:python3-typed-ast-1.3.1-bp151.2.6.1",
          "openSUSE Leap 15.1:python3-typed-ast-1.3.1-lp151.2.6.1"
        ]
      }
    ],
    "threats": [
      {
        "type": "Impact",
        "description": "moderate"
      }
    ],
    "cvss_score_sets": {
      "score_set_v_2": {
        "base_score_v_2": "5",
        "vector_v_2": "AV:N/AC:L/Au:N/C:N/I:N/A:P"
      },
      "score_set_v_3": {
        "base_score_v_3": "7.5",
        "vector_v_3": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
      }
    }
  }
}
